src/core/hle/kernel/svc.cpp

```diff
@@ -914,6 +917,17 @@ static ResultCode GetInfo(Core::System& system, u64* result, u64 info_id, Handle
         *result = system.Kernel().CurrentScheduler()->GetIdleThread()->GetCpuTime();
         return ResultSuccess;
     }
+    case GetInfoType::MesosphereCurrentProcess: {
+        R_UNLESS(handle == InvalidHandle, ResultInvalidHandle);
+        R_UNLESS(info_sub_id == 0, ResultInvalidCombination);
+
+        KProcess* const current_process = system.Kernel().CurrentProcess();
+        Handle process_handle{};
+        R_TRY(current_process->GetHandleTable().Add(&process_handle, current_process));
+
+        *result = process_handle;
+        return ResultSuccess;
+    }
```

Comment:
I would try to intentionally match this to the style used by Atmosphere:
```suggestion
        // Verify the input handle is invalid.
        R_UNLESS(handle == InvalidHandle, ResultInvalidHandle);

        // Verify the sub-type is valid.
        R_UNLESS(info_sub_id == 0, ResultInvalidCombination);

        // Get the handle table.
        KProcess* current_process = system.Kernel().CurrentProcess();
        KHandleTable& handle_table = current_process->GetHandleTable();

        // Get a new handle for the current process.
        Handle tmp;
        R_TRY(handle_table.Add(&tmp, current_process));

        // Set the output.
        *result = tmp;
        
        // We succeeded.
        return ResultSuccess;
```